pkg/sdk/go/users_test.go

```diff
@@ -128,6 +133,220 @@ func TestCreateUser(t *testing.T) {
+	}
+
+	var users2 []sdk.User
+	for i := 1; i < 5; i++ {
+		email := fmt.Sprintf("test3-%d@mainflux.com", i)
+		password := fmt.Sprintf("password%d", i)
+		metadata := map[string]interface{}{"name": "mainflux"}
+		us := sdk.User{Email: email, Password: password, Metadata: metadata}
+		userID, err := mainfluxSDK.CreateUser(token, us)
+		require.Nil(t, err, fmt.Sprintf("unexpected error: %s", err))
+		us.ID = userID
+		us.Password = ""
+		users2 = append(users2, us)
+	}
+
+	cases := []struct {
+		desc     string
+		token    string
+		offset   uint64
+		limit    uint64
+		err      error
+		response []sdk.User
+		email    string
+		metadata map[string]interface{}
+	}{
+		{
+			desc:   "get a list of users",
+			token:  token,
+			offset: offset,
+			limit:  limit,
+			err:    nil,
+			email:  email,
+		},
+		{
+			desc:   "get a list of users with invalid token",
+			token:  wrongValue,
+			offset: offset,
+			limit:  limit,
+			err:    createError(sdk.ErrFailedFetch, http.StatusUnauthorized),
+			email:  email,
+		},
+		{
+			desc:   "get a list of users with empty token",
+			token:  "",
+			offset: offset,
+			limit:  limit,
+			err:    createError(sdk.ErrFailedFetch, http.StatusUnauthorized),
+			email:  email,
+		},
+		{
+			desc:   "get a list of users with zero limit",
+			token:  token,
+			offset: offset,
+			limit:  0,
+			err:    createError(sdk.ErrFailedFetch, http.StatusBadRequest),
+			email:  email,
+		},
+		{
+			desc:   "get a list of users with limit greater than max",
+			token:  token,
+			offset: offset,
+			limit:  110,
+			err:    createError(sdk.ErrFailedFetch, http.StatusBadRequest),
+			email:  email,
+		},
+		{
+			desc:     "get a list of users with same email address and no metadata",
+			token:    token,
+			offset:   0,
+			limit:    100,
+			err:      nil,
+			email:    "mainflux.com",
+			metadata: make(map[string]interface{}),
+		},
+		{
+			desc:   "get a list of users with same email address and metadata",
+			token:  token,
+			offset: 0,
+			limit:  100,
+			err:    nil,
+			email:  "mainflux.com",
+			metadata: map[string]interface{}{
+				"name": "mainflux",
+			},
+		},
+		{
+			desc:   "get a list of users with same metadata and no email address",
+			token:  token,
+			offset: 0,
+			limit:  100,
+			err:    nil,
+			email:  "",
+			metadata: map[string]interface{}{
+				"name": "demo5",
+			},
+		},
+		{
+			desc:     "get a list of users with same no metadata and email address",
+			token:    token,
+			offset:   0,
+			limit:    100,
+			err:      nil,
+			email:    "mainflux.com",
+			metadata: make(map[string]interface{}),
+		},
+	}
+	for _, tc := range cases {
+		filter := sdk.PageMetadata{
+			Email:    tc.email,
+			Total:    uint64(200),
+			Offset:   uint64(tc.offset),
+			Limit:    uint64(tc.limit),
+			Metadata: tc.metadata,
+		}
+		_, err := mainfluxSDK.Users(tc.token, filter)
+		assert.Equal(t, tc.err, err, fmt.Sprintf("%s: expected error %s, got %s", tc.desc, tc.err, err))
```

Comment:
Do not only test errors but also test responses, similar to `TestUser`.

Reply:
For this, we were not checking the response because it doesn't return in the order they were created. For instance, you want to query 5 users, it will return random 5 users i.e 
```
Email:"test-65@example.com"
Email:"test-67@example.com"
Email:"test-75@example.com"
Email:"test-93@example.com"
Email:"test-94@example.com"
```

Instead of users 1 to 5

Reply:
Please don't forget this one.